app/components/providers/deckListProvider/DeckListProvider.tsx

```diff
@@ -1,14 +1,60 @@
 import {
   FC,
-  PropsWithChildren,
   createContext,
+  useEffect,
   useState,
   useContext,
   useReducer,
+  ReactNode,
 } from "react";
-import util from "../../util";
 import { CardDataContext } from "../cardDataProvider";
 import Swal from "sweetalert2";
+import { PocketBaseContext } from "../pocketBaseProvider/PocketBaseProvider";
+
+const parseCardIntoDeckCard = (cardObj: any, cardCount: number = 1) => {
+  return {
+    cost: cardObj.cost,
+    cardNum: cardObj.cardNum,
+    name: cardObj.name,
+    imageName: cardObj.deckCardImage,
+    count: cardCount,
+    isEpic: cardObj.type[0] === "✦",
+  };
+};
+
+type dbDeckListObjType = { [key: string]: string };
+
+async function getDeckList(id: string) {
+  const res = await fetch(
+    `https://grotto-beasts-test.fly.dev/api/collections/decklists/records/${id}`
+  );
+  const data = await res.json();
+  return data;
+}
+
+function createDeckListObj(deckList: dbDeckListObjType, cardsData: any) {
+  if (!cardsData) {
+    return [];
+  }
+  var deckListInit: DeckListType[] = [];
+  for (var cardNum in deckList) {
+    var cardObj = cardsData[parseInt(cardNum) - 1];
+    var cardCount = deckList[cardNum];
+    var parsedCard = parseCardIntoDeckCard(cardObj, parseInt(cardCount));
+    deckListInit.push(parsedCard);
+  }
+  return deckListInit;
+}
+
+const formatDeckObj = (currDeckObj: any) => {
+  var returnObj: { [key: number]: number } = {};
+  for (var card of currDeckObj) {
+    if (card.count > 0) {
+      returnObj[card.cardNum] = card.count;
+    }
+  }
+  return returnObj;
+};
```

Comment:
```suggestion
const formatDeckObj = (currDeckObj: any): { [key: number]: number } => {
  return currDeckObj.reduce((returnObj, { cardNum, count }) => {
    if (count > 0) {
      returnObj[cardNum] = count;
    }
    return returnObj;
  }, {});
};

```

"
We use the `reduce` function to iterate through the `currDeckObj` array and build the `returnObj` object.

The initial value of the reduce function is an empty object {}.

We directly assign properties to the `returnObj` object if the `card.count` is greater than 0.

We use object destructuring to directly extract `cardNum` and `count` from each `card` object within the `currDeckObj` array.

This makes the code more concise and self-explanatory by eliminating the need to access properties like `card.cardNum` and `card.count`.

This is a minor improvement that enhances the readability of your code without sacrificing performance or functionality."

Reply:
Done